src/qat/purr/integrations/qasm.py

```diff
@@ -1580,11 +1581,26 @@ def _validate_arg_length(arg_tree, *lengths):
             )
 
         elif intrinsic_name == "gaussian_square":
-            amp, width, square_width, std_dev = _validate_arg_length(tree.children[4], 4)
+            amp, width, square_width, std_dev, zero_at_edges = _validate_arg_length(
+                tree.children[4], 4, 5
+            )
+            zero_at_edges = 0 if not zero_at_edges else 1
```

Comment:
`zero_at_edges = int(zero_at_edges)`

minor: still looks like there's wavering between specification of `zero_at_edges` as int or boolean. I'd recommend to stick to one choice or at least hoist all conversion operations to one location (becomes easier to debug in case of issues)

Reply:
Yeah I agree, changed it to be processed as a bool from the start.

On this note, I think something for later is to relook / think how we are defining pulses. At both the level of parsing the QASM script and implementing the numerical function, there are many inconsistencies. I think this can overall be made to be a lot cleaner and easier to follow.